compiler/modules/front/stmt.bal

```diff
@@ -699,6 +703,12 @@ function codeGenMatchStmt(StmtContext cx, bir:BasicBlock startBlock, Environment
     else {
         // Match expression is not a variable: we do not get type narrowing
         int patternIndex = 0;
+        if matched is bir:TmpRegister {
+            bir:AssignTmpRegister result = cx.createAssignTmpRegister(matched.semType, matched.pos);
+            bir:AssignInsn insn = { result, operand: matched, pos : stmt.startPos };
+            matched = result;
```

Comment:
In a match we are only reading `matched`, aren't we ? why do we need `AssignTmpRegister`?

Reply:
In a case like 
```

Reply:
In a case like 
```
match x + 1 {
    2 => { 
        return "1";
    }
    4 => {
        return "2";
    }
}
```
even though matched is a `TmpRegister` it will be used in multiple blocks for Equality instructions, so to avoid that I think we can either do something like this or change all expr code gens to create `AssignTmpRegister` instead of Tmp depending on the situation.

Reply:
I think we'll have to change the definition of `TmpRegister` to, assigned once, but can be used multiple times.
@jclark what do you think?

Reply:
I believe this is already the case, the problem is when the `TmpRegister` is used in a different block than the block it was initially assigned.

Reply:
> but can be used multiple times
I meant
but can be used from multiple block
let me edit.